Mobile Buy SDK/Mobile Buy SDK/Data/BUYClient+Checkout.m

```diff
@@ -205,26 +119,81 @@ - (BUYRequestOperation *)getCompletionStatusOfCheckoutURL:(NSURL *)url completio
 	
 	BUYAssert(token, @"Failed to get completion status of checkout. Checkout URL must have a valid token associated with it.");
 	
-	return [self getCompletionStatusOfCheckoutToken:token start:YES completion:block];
+	return [self getCompletionStatusOfCheckoutWithToken:token start:YES completion:block];
 }
 
-- (BUYRequestOperation *)getCompletionStatusOfCheckoutToken:(NSString *)token start:(BOOL)start completion:(BUYDataStatusBlock)block
+#pragma mark - Checkout Helpers -
+
+- (BUYRequestOperation *)beginCheckoutWithToken:(NSString *)checkoutToken paymentToken:(id<BUYPaymentToken>)paymentToken completion:(BUYDataCheckoutBlock)block
+{
+	BUYAssertToken(checkoutToken);
+	
+	NSURL *route = [self urlForCheckoutsCompletionWithToken:checkoutToken];
+	return [self postRequestForURL:route object:[paymentToken JSONDictionary] start:NO completionHandler:^(NSDictionary *json, NSHTTPURLResponse *response, NSError *error) {
+		[self handleCheckoutResponse:json error:error block:block];
+	}];
+}
+
+- (BUYRequestOperation *)getCompletionStatusOfCheckoutWithToken:(NSString *)token start:(BOOL)start completion:(BUYDataStatusBlock)block
 {
 	NSURL *url = [self urlForCheckoutsProcessingWithToken:token];
 	return [self getRequestForURL:url start:start completionHandler:^(NSDictionary *json, NSHTTPURLResponse *response, NSError *error) {
 		block([self statusForStatusCode:response.statusCode error:error], error);
 	}];
 }
 
+- (BUYRequestOperation *)postCheckout:(NSDictionary *)checkoutJSON completion:(BUYDataCheckoutBlock)block
+{
+	return [self postRequestForURL:[self urlForCheckouts] object:checkoutJSON completionHandler:^(NSDictionary *json, NSHTTPURLResponse *response, NSError *error) {
+		[self handleCheckoutResponse:json error:error block:block];
+	}];
+}
+
+- (void)configureCheckout:(BUYCheckout *)checkout
+{
+	checkout.marketingAttribution = @{@"medium": @"iOS", @"source": self.applicationName};
+	checkout.sourceName = @"mobile_app";
+	if (self.urlScheme || checkout.webReturnToURL) {
+		checkout.webReturnToURL = checkout.webReturnToURL ?: [NSURL URLWithString:self.urlScheme];
+		checkout.webReturnToLabel = checkout.webReturnToLabel ?: [@"Return to " stringByAppendingString:self.applicationName];
+	}
+}
+
+- (void)handleCheckoutResponse:(NSDictionary *)json error:(NSError *)error block:(BUYDataCheckoutBlock)block
+{
+	BUYCheckout *checkout = nil;
+	if (json && !error) {
+		checkout = [self.modelManager insertCheckoutWithJSONDictionary:json[@"checkout"]];
+	}
+	block(checkout, error);
+}
+
+- (void)updateCheckout:(BUYCheckout *)checkout withGiftCardDictionary:(NSDictionary *)giftCardDictionary addingGiftCard:(BOOL)addingGiftCard
```

Comment:
I think we should break up this method. We can extract the identical logic to another method, but it's only two lines.